src/main/java/it/pagopa/pn/delivery/rest/PnNotificationPriceController.java

```diff
@@ -39,7 +39,7 @@ public ResponseEntity<NotificationPriceResponse> getNotificationPrice(String paT
             logEvent.generateSuccess().log();
         } catch (PnNotFoundException exc) {
             logEvent.generateFailure("Exception on get notification price: " + exc.getMessage()).log();
-            throw new RuntimeException("Exception on get notification price: " + exc.getMessage(), exc);
+            throw new PnNotFoundException("Exception on get notification price: " + exc.getMessage(), exc);
         }
```

Comment:
Qui fa la catch di PnNotFoundException e va rilanciata la stessa, no importa crearne una nuova. Ma ci sono anche le altre Exception che devono comunque finire nell'audit log.

Reply:
Qui mi sono ispirato alla documentazione IBM https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions sulle best-practice, in particolare il punto 4.

> The correct way to catch and re-throw an exception is to pass the caught exception object as the "rootCause" or inner exception parameter to the constructor of the new exception (note that not all exception constructors support inner exceptions, in which case a different wrapper exception should be used). When the exception is later caught and printed to SystemOut.log, the inner exception message and stack will be included: